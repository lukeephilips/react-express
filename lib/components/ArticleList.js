import React from 'react';
import Article from './Article';

class ArticleList extends React.PureComponent {
  render() {
    const listArticles = Object.values(this.props.articles).map((article) =>
      <Article
        article={article}
        key={article.id}
      />
    );
    return (
      <div>
        {listArticles}
      </div>
    );
  }
}

export default ArticleList;
