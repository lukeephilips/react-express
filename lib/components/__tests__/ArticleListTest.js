import React from 'react';
import ArticleList from './../ArticleList';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
describe('ArticleList', () => {
  const testProps = {
    articles: {
      a: {
        'id': 'a',
        'title': 'Thing',
        'date': 'Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)',
        'authorId': '2c6aa2cfe3449467d329fa17d6ea230f',
        'body': 'blah blah'
      },
      b: {
        'id': 'b',
        'title': 'Ding',
        'date': 'Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)',
        'authorId': '2c6aa2cfe3449467d329fa17d6ea230f',
        'body': 'blah blah'
      },
    },
  };

  it('renders correctly', () => {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('ArticleContainer').length).toBe(2);
  });
});
