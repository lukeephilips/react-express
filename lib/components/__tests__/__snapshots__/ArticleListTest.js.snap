// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleList
    articles={
      Object {
        "a": Object {
          "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
          "body": "blah blah",
          "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
          "id": "a",
          "title": "Thing",
        },
        "b": Object {
          "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
          "body": "blah blah",
          "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
          "id": "b",
          "title": "Ding",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Article
          article={
            Object {
              "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
              "body": "blah blah",
              "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
              "id": "a",
              "title": "Thing",
            }
          }
          store={undefined}
        />,
        <Article
          article={
            Object {
              "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
              "body": "blah blah",
              "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
              "id": "b",
              "title": "Ding",
            }
          }
          store={undefined}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "a",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
            "body": "blah blah",
            "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
            "id": "a",
            "title": "Thing",
          },
          "store": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "b",
        "nodeType": "function",
        "props": Object {
          "article": Object {
            "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
            "body": "blah blah",
            "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
            "id": "b",
            "title": "Ding",
          },
          "store": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Article
            article={
              Object {
                "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
                "body": "blah blah",
                "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
                "id": "a",
                "title": "Thing",
              }
            }
            store={undefined}
          />,
          <Article
            article={
              Object {
                "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
                "body": "blah blah",
                "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
                "id": "b",
                "title": "Ding",
              }
            }
            store={undefined}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "a",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
              "body": "blah blah",
              "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
              "id": "a",
              "title": "Thing",
            },
            "store": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "b",
          "nodeType": "function",
          "props": Object {
            "article": Object {
              "authorId": "2c6aa2cfe3449467d329fa17d6ea230f",
              "body": "blah blah",
              "date": "Mon Jul 11 2016 00:00:00 GMT+0000 (UTC)",
              "id": "b",
              "title": "Ding",
            },
            "store": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
